### ISSUE TITLE
Fast channel bringup

### ISSUE THREAD

**Issue by liorzilp** [dymensionxyz/dymension#233]
## Overview

In order to open a channel between two chains, there are many steps that need to be done, such as creating a client on each endpoint, creating a connection, and creating a channel. Each step is a process that requires 3-4 messages, the total delay is at least 6 fold of the finalization-time.

## Problem

In cases where the finalization time is days, bringing up a connection could take weeks.

## Suggested solution

Don't wait for finalization in order to bring-up the channel. Make it immediate.
Postpone the processing of the last message of the handshake (ChanOpenIConfirm) to be handled only when it becomes finalized.
If there wasn't rollback since the start of the process, the channel will become open and could be used.
If there was a rollback, mark the client as frozen and no updates could be applied to it, so a new client should be used for creating a new connection (see Client update state).
It will make the bring-up process time to be as long as 1 finalization time.

### IRC requests on a channel in fast-bringup process
IBC messages can be sent on a channel that is not open on both sides, for example, the rollapp side is open, but the hub side is not (waiting for the finalization of the last message). Even though we could queue the message (as [IRC request](https://github.com/dymensionxyz/dymension/issues/202) ) we will take precautions and fail the message. This scenario can be only when opening a new channel, so it is not really changing the uX.

### Client update state
The client is the basic object for handling any channel message as well as handshake messages.
In order to process handshake messages without waiting for finalization, the client state updates should skip the finalization time (because the handshake messages are verified against the client state).
Skipping finalization is secure as long as any other messages, besides the handshake message, couldn't be sent on any channel connected to this client. Therefore, this mechanism is safe to use only for the first channel bring-up between a specific client. Otherwise, a typical attack will be to create a new connection/channel on the same client making it updated by non-finalized states, which might be malicious.
In other words, the client update pass-through mode is global to the client and can be applied only once for a single channel.
Any new channel that will be opened on this client will take the full route.

**Comment by mtsitrin** [comment]
closed with #356
